### PR TITLE
feat: added hostPID capability for linux daemonset

### PR DESCRIPTION
**Pull request by jamesalford**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind flake

/kind feature

**What this PR does / why we need it**:
Adds the ability to add hostPID to the container spec for linux daemonset. This is used to execute scripts in init containers that modify the host storage

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note

```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

File: helm/provisioner/templates/daemonset_linux.yaml [hunk 1]
```diff
@@ -28,6 +28,9 @@ spec:
 {{ .Values.podAnnotations | toYaml | trim | indent 8 }}
 {{- end }}
     spec:
+{{- if .Values.hostPID }}
+      hostPID: true
```
Comment (mauriciopoppe) [inline helm/provisioner/templates/daemonset_linux.yaml:32]: Nit: Maybe this should be:

```
hostPID: {{ .Values.hostPID }}
```

The default value in the reference is `false` https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#podspec-v1-core, also the value set from the chart is `false` so I think the above would work.

File: helm/README.md [hunk 2]
```diff
@@ -231,6 +231,7 @@ provisioner chart and their default values.
 | serviceMonitor.namespace                | The namespace Prometheus servicemonitor will be installed                                                                      | str      | `.Release.Namespace`                                          |
 | serviceMonitor.additionalLabels         | Additional labels for the servicemonitor                                                                                       | map      | `-`                                                           |
 | serviceMonitor.relabelings              | Additional metrics relabel_config                                                                                              | lists    | `-`                                                           |
+| hostPID                                 | Add `hostPID: true` to the container spec for linux daemonset.                                                                 | bool     | `false`                                                       |
```
Comment (mauriciopoppe) [inline helm/README.md:234]: If you consider the comment below then the docs should be updated too.

**Pull request closed without merging**